site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -9164,20 +9164,23 @@ private void actionForTemplate(String direction, int index,
 							final String userEmail = UserDirectoryService.getCurrentUser().getEmail();
 							final Session session = SessionManager.getCurrentSession();
 							final ToolSession toolSession = SessionManager.getCurrentToolSession();
+							final String siteId = existingSite.getId();
 							Thread siteImportThread = new Thread(){
 								public void run() {
-									Site existingSite = getStateSite(state);
-									EventTrackingService.post(EventTrackingService.newEvent(SiteService.EVENT_SITE_IMPORT_START, existingSite.getReference(), false));
-									SessionManager.setCurrentSession(session);
-									SessionManager.setCurrentToolSession(toolSession);
-									importToolIntoSite(existingTools, importTools, existingSite);
-									existingSite = getStateSite(state); // refresh site for
-									// WC and News
-									commitSite(existingSite);
-									if (ServerConfigurationService.getBoolean(SAK_PROP_IMPORT_NOTIFICATION, true)) {
-										userNotificationProvider.notifySiteImportCompleted(userEmail, existingSite.getId(), existingSite.getTitle());
+									Site existingSite;
+									try {
+										existingSite = SiteService.getSite(siteId);
+										EventTrackingService.post(EventTrackingService.newEvent(SiteService.EVENT_SITE_IMPORT_START, existingSite.getReference(), false));
```

Comment:
I think the title of this jira is the confusion point. It's not technically removing the whole session state, its remove the AssignmentAction session state (e.g. current site, etc). Here's the whole thread, where you can see that the user session is being set still:


Thread siteImportThread = new Thread(){
								public void run() {
									Site existingSite;
									try {
										existingSite = SiteService.getSite(siteId);
										EventTrackingService.post(EventTrackingService.newEvent(SiteService.EVENT_SITE_IMPORT_START, existingSite.getReference(), false));
										SessionManager.setCurrentSession(session);
										SessionManager.setCurrentToolSession(toolSession);
										importToolIntoSite(existingTools, importTools, existingSite);
										if (ServerConfigurationService.getBoolean(SAK_PROP_IMPORT_NOTIFICATION, true)) {
											userNotificationProvider.notifySiteImportCompleted(userEmail, existingSite.getId(), existingSite.getTitle());
										}
										EventTrackingService.post(EventTrackingService.newEvent(SiteService.EVENT_SITE_IMPORT_END, existingSite.getReference(), false));
									} catch (IdUnusedException e) {
										M_log.error(e.getMessage(), e);
									}
								}
							};